interfaces/builtin/mir.go

```diff
@@ -0,0 +1,606 @@
+
+unix (receive, send) type=seqpacket addr=none,
+/tmp/mir_socket rw,
+/dev/dri/card0 rw,
+/dev/shm/\#* rw,
+
+/sys/devices/**/uevent rw,
+/sys/devices/**/ r,
+/dev/input/* rw,
+/dev/tty* wr,
+/run/udev/data/* r,
+/run/udev/** rw,
+
+ptrace peer=**,
+/bin/sleep mrix,
+/bin/pidof mrix,
+/bin/sed mrix,
+/bin/cp mrix,
+/sbin/killall5 ixr,
+/usr/bin/expr ixr,
+/usr/bin/chmod ixr,
+/bin/chmod ixr,
+/proc/ r,
+/proc/*/stat r,
+/proc/*/cmdline r,
+/sys/bus/ r,
+/sys/class/ r,
+/sys/class/input/ r,
+/sys/class/drm/ r,
+/etc/udev/udev.conf r,
+capability sys_ptrace,
+capability chown,
+capability fowner,
+capability sys_ptrace,
+`)
+
+var mirConnectedPlugSecComp = []byte(`
+# Description: Allow operating as the mir service. Reserved because this
+# gives privileged access to the system.
+
+`)
+
+type MirInterface struct{}
+
+func (iface *MirInterface) Name() string {
+	return "mir"
+}
+
+func (iface *MirInterface) PermanentPlugSnippet(
+            plug *interfaces.Plug, 
+            securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem{
+        case interfaces.SecurityAppArmor:
+                return mirPermanentSlotAppArmor, nil
+        case interfaces.SecuritySecComp:
+                return mirPermanentSlotSecComp, nil
+	case interfaces.SecurityUDev, interfaces.SecurityDBus:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *MirInterface) ConnectedPlugSnippet(
+            plug *interfaces.Plug, 
+            slot *interfaces.Slot, 
+            securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return nil, nil
+	case interfaces.SecuritySecComp:
+		return nil, nil
+	case interfaces.SecurityUDev, interfaces.SecurityDBus:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *MirInterface) PermanentSlotSnippet(
+            slot *interfaces.Slot, 
+            securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return nil, nil
+	case interfaces.SecuritySecComp:
+		return nil, nil
+	case interfaces.SecurityUDev, interfaces.SecurityDBus:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *MirInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+        case interfaces.SecurityAppArmor:
+		return nil, nil
+        case interfaces.SecuritySecComp:
+		return nil, nil
+        case interfaces.SecurityUDev, interfaces.SecurityDBus:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *MirInterface) SanitizePlug(plug *interfaces.Plug) error {
+	return nil
+ }
+
+func (iface *MirInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	return nil
+}
+
+func (iface *MirInterface) AutoConnect() bool {
```

Comment:
We've said the slot side should autoconnect, yes, and the plugs side of Mir is supposed to be safe for anyone, so yes, autoconnect.